pydantic/networks.py

```diff
@@ -54,7 +54,7 @@ def url_regex() -> Pattern[str]:
     if _url_regex_cache is None:
         _url_regex_cache = re.compile(
             r'(?:(?P<scheme>[a-z][a-z0-9+\-.]+)://)?'  # scheme https://tools.ietf.org/html/rfc3986#appendix-A
-            r'(?:(?P<user>[^\s:/]+)(?::(?P<password>[^\s/]*))?@)?'  # user info
+            r'(?:(?P<user>[^\s:/]*)(?::(?P<password>[^\s/]*))?@)?'  # user info
```

Comment:
this is quite a significant change - allow passing without user.

I think it's a good idea but it needs:
* to be noted in documentation
* to have it's own item in change log
* to have tests covering this case.

Reply:
these items still need resolving.

Reply:
@TrDex this can't move forward until you act upon this comment.